### DISCUSSION TITLE
Use templates in allowedVersion pattern

### DISCUSSION THREAD

**Discussion by sams-gleb** [renovatebot/renovate#20722] [Request Help]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, select which platform you are using.

Bitbucket Server

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

is it possible to use current version as part of `allowedVersions`? we have a use case where we want to update automatically to all new patch versions and minor+1, not further. Would be helpful if we could render templates in allowed version patterns e.g. `"allowedVersions": "/^[0-9]+.({{minor}}|{{minor+1}}).[0-9]+$/"`or maybe there are some other possibilities?

### What you tried so far.

use static values in the `allowedVersions` pattern in renovate.json and update them after releasing new version

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by rarkins** [comment]
It's not currently supported, but it's an interesting idea. I don't think your syntax of `{{minor+1}}` is supported in handlebars templating though.

What are you originally trying to achieve?

**Reply by sams-gleb** [reply]
I'm trying to limit the minor version we can upgrade to. In our workflow it should be possible to update only to one next minor version, so if we have artifact with "currentVersion.minor+1" it should be eligible for auto updates

**Reply by rarkins** [reply]
So you mean you'd limit yourself to only upgrading to the next minor each time, and ignoring all future majors and minor releases, even if you're multiple majors and minors behind?

**Reply by sams-gleb** [reply]
yes, that's what I want to achieve, update to a limited number of minor versions and ignore newer ones

**Reply by rarkins** [reply]
You would first need to research how to do math using Handlebars templates

**Comment by superjo007** [comment]
Same here, we have a usecase where the templating would help when using release branches.
We only want versions similar to the release branch. currentVersion can be 4.*.* but we want renovate to stick to the release version only.

example: branch is release/5.0

allowedVersions: '{{{replace 'release/' '' baseBranch}}}' ==> allowedVersions: '5.0'

Cheers.
